### PR TITLE
ft(#4): Reset password

### LINKED ISSUE #4: Implement Reset pasword
- As a registered trainee, I should be able to forgot and change my password
- Appropriate error should be displayed

### PR DESCRIPTION
**Pull request by munezeromicha**
### PR Description

This pull request implements a new feature that allows users to reset their passwords by sending a password reset email. When a user clicks the reset password link in the email, the app will open and redirect them to the setNewPassword screen.

### Description of tasks that were expected to be completed

- Create a new password reset endpoint: Implement a new endpoint that handles password reset requests. This endpoint should generate a unique password reset token and send an email to the user containing a link to a password reset page.
- Implement password reset functionality: Create a new screen or page where users can enter their new password. When a user submits the new password, the app should verify the password reset token and update the user's password if the token is valid.
- Handle email delivery: Integrate an email service (e.g., SendGrid, Mailgun) to send password reset emails. Ensure that the email template includes a clear link to the password reset page.
- Implement security measures: Implement security measures to protect password reset tokens and prevent unauthorized password resets. Consider using techniques like token expiration, token invalidation, and rate limiting.

### How has this been tested?

1. Create a new user account.
2. Request a password reset.
3. Click the password reset link in the email.
4. Enter a new password on the setNewPassword screen.
5. Verify that the password reset is successful by logging in with the new password.
6. Test the password reset functionality with invalid email addresses, expired tokens, and rate limiting.

### Track PR 
N/A

### Screenshots:
<div style="display: flex; flex-wrap: wrap;">
    Screenshot 1
        Screenshot 1
    Screenshot 2
        Screenshot 1
</div>

### CODE REVIEW

**Review by pacifiquemboni — APPROVED** [review]

**Review by Mbabazi12 — APPROVED** [review]

File: app/(onboarding)/index.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ export default function AppOnboarding() {
       </View>
       <View className={`flex-1 flex-row justify-center items-center ${bgColor}`}>
         <TouchableOpacity>
-          <Text className="text-lg font-Inter-Medium" onPress={() => router.push('/auth/login')}>
+          <Text className={`text-lg font-Inter-Medium ${textColor}`} onPress={() => router.push('/auth/login/login')}>
```
Comment (ncutixavier) [inline app/(onboarding)/index.tsx]: This route is not well structured. Let's find a way to keep "/auth/login"

File: app/auth/forget/reset-password.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            toast.show(err.message, {
+              type: 'danger',
+              placement: 'top',
+              duration: 4000,
+              animationType: 'slide-in',
+            });
+          },
+        });
+      } catch (err) {
+        console.log('Caught error:', err);
```
Comment (ncutixavier) [inline app/auth/forget/reset-password.tsx]: Remove all logs

**Review by ncutixavier — CHANGES_REQUESTED** [review]

File: app/auth/forget/reset-password.tsx (hunk 2 above)
Comment (aimeudoxie) [reply app/auth/forget/reset-password.tsx]: Done

File: app/(onboarding)/index.tsx (hunk 1 above)
Comment (aimeudoxie) [reply app/(onboarding)/index.tsx]: Done

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**